Updates to reflect updates made to Application and Tenant UIs.

bguyza opened the pull request:
https://github.com/FusionAuth/fusionauth-internal-issues/issues/126

bguyza commented on site/docs/v1/tech/connectors/index.adoc:
```diff
@@ -3,6 +3,7 @@ layout: doc
 title: FusionAuth Connectors
 description: FusionAuth Identity Providers
 ---
+:page-liquid:
```
Fixes some formatting.

robotdan commented on site/docs/v1/tech/core-concepts/applications.adoc:
```diff
@@ -223,12 +223,18 @@ The issuer used by service providers (i.e. Google, Zendesk, etc.) to identify th
 Some service providers require a different audience (such as Zendesk). You can leave this blank if the audience is the same as the issuer.
 
-[field]#Callback URL (ACS)# [required]#Required#::
+[field]#Authorized redirect URLs (ACS)# [required]#Required#::
```
I think the "ACS" should be explained int he field description, so we can remove `(ACS)` from the field name here now that have renamed it to `Authorized redirect URLs`.

robotdan commented on site/docs/v1/tech/core-concepts/applications.adoc:
```diff
@@ -223,12 +223,18 @@ The issuer used by service providers (i.e. Google, Zendesk, etc.) to identify th
 
-[field]#Callback URL (ACS)# [required]#Required#::
+[field]#Authorized redirect URLs (ACS)# [required]#Required#::
 The URL that FusionAuth returns the user to after they have logged in via SAML v2, also known as the Assertion Consumer Service URL (ACS).
```
This is now a plural configuration.

robotdan commented on site/docs/v1/tech/core-concepts/applications.adoc:
```diff
@@ -223,12 +223,18 @@ The issuer used by service providers (i.e. Google, Zendesk, etc.) to identify th
+Enable debug to create an event log to assist you in debugging integration errors.
+
+[field]#Require signature# [optional]#Optional#::
+Enable to required the SAML v2 Service Provider to sign the SAML v2 authentication request. When enabled, if a signature is omitted, the request will be rejected.
```
Grammar. 

> Enable to require the SAMl v2 ...


Suggestion:

> When enabled, all unsigned requests will be rejected.

robotdan commented on site/docs/v1/tech/core-concepts/applications.adoc:
```diff
@@ -223,12 +223,18 @@ The issuer used by service providers (i.e. Google, Zendesk, etc.) to identify th
+
+[field]#Require signature# [optional]#Optional#::
+Enable to required the SAML v2 Service Provider to sign the SAML v2 authentication request. When enabled, if a signature is omitted, the request will be rejected.
+
```
Missing "Default verification key" field.

bguyza replied:
Let me know if you approve of the way I structured the description for this.

robotdan commented on site/docs/v1/tech/core-concepts/applications.adoc:
```diff
@@ -290,6 +293,14 @@ The optional fields to be displayed on the registration form.
+
+[.api]
+[field]#Form# [optional]#Optional#::
+The Admin form used by FusionAuth for adding or editing a registration. FusionAuth provides a default form that will be used unless otherwise specified.
```
> The form that will be used in the FusionAuth UI for adding and editing user registrations.

robotdan commented on site/docs/v1/tech/core-concepts/tenants.adoc:
```diff
@@ -53,6 +54,21 @@ The named issuer used to sign tokens.  Typically a fully-qualified domain name.
+
+[.api]
+[field]#Admin user form# [optional]#Optional#::
+The Admin form used by FusionAuth for adding or editing users. FusionAuth provides a default form that will be used unless otherwise specified.
```
> The form that will be used in the FusionAuth UI for adding and editing users.

robotdan commented on site/docs/v1/tech/core-concepts/tenants.adoc:
```diff
@@ -53,6 +54,21 @@ The named issuer used to sign tokens.  Typically a fully-qualified domain name.
+
+Connectors can be enabled on a per tenant basis with a Connector policy.
+
+image::connectors/tenant-connector-tab.png[The Tenant Connector policy configuration tab.,width=1200,role=shadowed]
```
We should probably document the add connector policy dialog here. External JWT IdP has an example of a dialog.

robotdan reviewed (commented):
Many of these images are larger in size than previously, did you follow all of the steps in the `DocsDevREADME`?

Also some of these look like they were manually copped for width?

Make sure all images are captured w/ dark mode off. This should be part of the screen size apple script which will ensure all images are the same size and dimension, etc.

robotdan commented on site/docs/v1/tech/core-concepts/tenants.adoc:
```diff
@@ -304,16 +320,15 @@ image::tenant-configuration-webhooks-settings.png[Tenant Configuration - JWT,wid
-[field]#Enabled# [optional]#Optional#::
+Event type::
+The event type, this value will be present in the JSON request to identify the message.
+Enabled::
 When enabled this event can be sent by one or more webhook. You will also need to enable the event for a specific webhook to receive the event.
 +
 This toggle allows you to optionally disable an event for all webhooks all at once.
-
-[field]#Transaction setting# [optional]#Optional#::
+Transaction setting::
 The transaction setting for this event. This setting will apply to all webhooks consuming this event type.
++
 +
```
I just corrected a similar issue over in webhooks, see https://github.com/FusionAuth/fusionauth-site/commit/fddb083ae51b9d123a287ad81c3dbeb0d2db9548 for an example. This formatting should be similar where we have an embedded table.

robotdan commented on site/docs/v1/tech/core-concepts/applications.adoc:
```diff
@@ -223,12 +223,23 @@ The issuer used by service providers (i.e. Google, Zendesk, etc.) to identify th
+[field]#Debug enabled# [optional]#Optional#::
+Enable debug to create an event log to assist you in debugging integration errors.
+
+[field]#Require signature# [optional]#Optional#::
+When enabled, all unsigned requests will be rejected.
+
+[field]#Default verification key# [optional]#Optional#::
```
needs `since`

robotdan commented on site/docs/v1/tech/core-concepts/applications.adoc:
```diff
@@ -290,6 +298,14 @@ The optional fields to be displayed on the registration form.
+==== Form settings
+
+[.api]
+[field]#Form# [optional]#Optional#::
```
Needs `since`

robotdan commented on site/docs/v1/tech/core-concepts/tenants.adoc:
```diff
@@ -53,6 +54,44 @@ The named issuer used to sign tokens.  Typically a fully-qualified domain name.
+===== Form Settings
+
+[.api]
+[field]#Admin user form# [optional]#Optional#::
```
Needs `since`

robotdan commented on site/docs/v1/tech/core-concepts/tenants.adoc:
```diff
@@ -53,6 +54,44 @@ The named issuer used to sign tokens.  Typically a fully-qualified domain name.
+[field]#Connector# [required]#Required#::
+The Connector to be used for this policy.
+
+[field]#Domains# [optional]#Optional#::
```
This field has a default if not specified. See https://fusionauth.io/docs/v1/tech/apis/tenants/#create-a-tenant

robotdan reviewed (commented):
Some of your screen shots are in dark mode and need to be re-captured using normal light mode. If you use the apple script, it should correct this for you. 

And a minor point, but on the `application-saml-enabled.png` screenshot, try to clip the top so the avatar isn't in the picture there is a cleaner line on the top.

bguyza commented:
@robotdan pushed to update per your last set of comments.

robotdan commented on site/docs/v1/tech/core-concepts/applications.adoc:
```diff
@@ -223,12 +223,23 @@ The issuer used by service providers (i.e. Google, Zendesk, etc.) to identify th
+[field]#Require signature# [optional]#Optional#::
+When enabled, all unsigned requests will be rejected.
+
+[field]#Default verification key# [optional]#Optional# [since]#Available since 1.20.0#::
```
Required when `[field]#Require signature#` is `true`.

robotdan reviewed (commented):
Something is still wrong with your screenshot procedures. 

It looks like you are cutting off the bottom of your images. 

- `/application-registration.png` is cut off at the bottom. 
- `/application-saml-enabled.png` is cut off at the bottom. 


Are you using the apple script to get the dimensions set correctly and then capturing w/out any modifications to the window size? 


- `- `/application-registration.png` is not correct, the old screen shot had configuration enabled so the screen shot was documenting the fields we are describing. 

If you aren't already, be sure to run this locally and read through it for consistency and correct placement etc. using `https://site-local.fusionauth.io/`.

robotdan commented:
A bunch of your screenshots still are not correct. Let me know if you need a hand understanding how to capture them correctly.

mooreds commented:
Note that I had to do a couple of things to get screenshots correct:

* take the screenshot on my mac, not on my LCD display
* go to system preferences -> displays -> display. Then choose resolution 'scaled' and 'more space'.

bguyza commented:
Thanks @mooreds. Yes, I did figure that out as the dimensions of my screenshots were wrong unless I used my laptop.  

@robotdan 
My process has been:
1.  Open browser using apple script.  Note: using
```
set appHeight to 1100
set appWidth to 1080
```
2. Open in preview app
3. Crop if necessary (only top/bottom)
4. Adjust width to 1600
5. Compress via tinypng.com and download

I was simply replacing the photos that were existing in the doc.  On some of those I did crop, so should I add the appropriate class to the image?

Other than that, please let me know if I'm missing something?

robotdan commented on site/docs/v1/tech/core-concepts/applications.adoc:
```diff
@@ -223,12 +223,26 @@ The issuer used by service providers (i.e. Google, Zendesk, etc.) to identify th
+* When using HTTP Redirect bindings.
+
++
+This field is required when `Require signature` is enabled.
```
Use field syntax here (see my previous comment)

robotdan reviewed (commented):
Your screen shots are still messed up, not sure what the deal is. 

Also, while you're int his doc, it is good to do a quick audit, the JWT tab is no longer in sync, can you update it?

robotdan commented:
Let me know when this one is ready for review, there are still un-resolved comments.

bguyza commented:
@robotdan 
Pushed updates to Application -> JWT section and resolved comment requesting use of field syntax.  
Let me know if the updated screen shot in the JWT section is acceptable? There is limited real estate on that section now so I was trying to capture the important content.

robotdan commented on site/docs/v1/tech/core-concepts/applications.adoc:
```diff
@@ -223,12 +230,26 @@ The issuer used by service providers (i.e. Google, Zendesk, etc.) to identify th
+* When the public key cannot be determined by the `KeyInfo` element when using POST bindings.
+* When using HTTP Redirect bindings.
+
++
```
Extra `+`?

bguyza replied:
Originally, the plus dropped the indent back. Now removed in favor of the text from the API.

robotdan commented on site/docs/v1/tech/core-concepts/applications.adoc:
```diff
@@ -223,12 +230,26 @@ The issuer used by service providers (i.e. Google, Zendesk, etc.) to identify th
+
+[field]#Default verification key# [optional]#Optional# [since]#Available since 1.20.0#::
+The default key used to verify the signature in the following scenarios:
+* When the public key cannot be determined by the `KeyInfo` element when using POST bindings.
+* When using HTTP Redirect bindings.
```
Here is the text from the API

> The verification key used to verify a signature when the SAML v2 Service Provider is using HTTP Redirect Bindings. 
> 
> When HTTP POST Bindings are used, this is the default verification key used if a `<KeyInfo>` element is not found in the SAML AuthNRequest. If a `<KeyInfo>` element is found, Key Master will be used to resolve the key and this configuration will not be used to verify the request signature. 
```

bguyza replied:
Got it. I replaced with this text.

robotdan reviewed (commented):
1. Why is `application-jwt-enabled.png` cropped on the bottom, doesn't looks like this saves much?
2. This screenshot is not correct `application-registration.png` I think I already mentioned this once before? 
3. Fix screenshots, should be using `local.fusionauth.io` instead of `localhost`.

bguyza commented:
@robotdan 
1.  All new screenshots (local.fusionauth.io)
* jwt-enabled is not cropped at the bottom. I captured all of the relevant info that I could fit within the browser window cropping the top.
* fixed application-registration.
2. Modified text for verification key.

robotdan commented:
I'm going to finish this one up, there are some new stuff that needs screenshots I think. I'll post my commit here.

robotdan commented:
@bguyza for reference, here is what I added to the Application Core concept.
https://github.com/FusionAuth/fusionauth-site/commit/c7306a57ea0782ab64d5e99228b430399d59c31a

- Added missing email tab
- Updated a bunch of old screenshots
- Added advanced and basic registration configuration forms
- Reordered doc to be consistent with the UI

When we touch these larger doc pages, this is a good opportunity to audit them and see what has fallen out of sync with the UI, new fields, new patterns for layout, etc.
